src/ValidationManifold.jl

```diff
@@ -107,16 +107,16 @@ function convert(::Type{ValidationTVector{V}}, X::V) where {V<:AbstractArray{<:N
     return ValidationTVector{V}(X)
 end
 
-function copyto!(p::ValidationMPoint, q::ValidationMPoint)
-    copyto!(p.value, q.value)
+function copyto!(M::ValidationManifold, p::ValidationMPoint, q::ValidationMPoint)
+    copyto!(M.manifold, p.value, q.value)
```

Comment:
```suggestion
    copyto!(M.manifold, p.value, q.value)
```
Shouldn't this check that `q` is a correct point? Also maybe we should be consistent in argument naming with other methods and swap the names of `p` and `q`?

Reply:
Oh, naming is inconsistent here, yes.

I think it should actually check both (input p and output q) since if you are looking for errors one might be that someone did a mistake in implementing `copyto!`?

Reply:
Checking the output doesn't necessarily make sense when it was just allocated. We could just check for the correct shape but we don't have any methods for that yet.
And this method also checks that someone assigns a point here, and not for example a tangent vector to a point.

Reply:
but `copyto!` copies the values so just before `return q` is should be a valid point on the manifold.

Reply:
Right, I didn't think about that. We could check that as well :slightly_smiling_face: .